services/metadactyl-clj/src/metadactyl/persistence/oauth.clj

```diff
@@ -117,10 +116,10 @@
   (let [id  (if (string? id) (UUID/fromString id) id)
         req (get-authorization-request id)]
     (when (nil? req)
-      (throw+ {:error_code ce/ERR_BAD_REQUEST
-               :reason     (str "authorization request " (str id) " not found")}))
+      (throw+ {:type  :clojure-commons.exception/not-found
```

Comment:
I think a 400 is right in this case, although it could be argued that a 404 is reasonable as well. When metadactyl prepares to do an OAuth redirection for Agave, it stores an authorization request object in the database. This ensures that metadactyl only accepts callbacks that it requested and it allows metadactyl to store some state information. This error can happen if the callback from an Agave authorization redirection returns an authorization ID that metadactyl doesn't recognize.